presto-main/src/main/java/io/prestosql/util/FinalizerService.java

```diff
@@ -37,7 +35,6 @@
 {
     private static final Logger log = Logger.get(FinalizerService.class);
 
-    private final Set<FinalizerReference> finalizers = Sets.newConcurrentHashSet();
```

Comment:
This queue doesn't seem to be used and I hope it doesn't also break the finalizer mechanism (since it keeps references to `FinalizerReference`).

Reply:
The `PhantomReference` [must remain referenced](https://docs.oracle.com/javase/8/docs/api/java/lang/ref/package-summary.html) until it is enqueued:

> The relationship between a registered reference object and its queue is one-sided. That is, a queue does not keep track of the references that are registered with it. If a registered reference becomes unreachable itself, then it will never be enqueued. It is the responsibility of the program using reference objects to ensure that the objects remain reachable for as long as the program is interested in their referents.

Reply:
Thank you! I didn't know that.